### ISSUE TITLE
DOC: fix GL08 errors in docstrings 

### ISSUE THREAD

**Issue by jordan-d-murphy** [pandas-dev/pandas#57443]
Pandas has a script for validating docstrings:

https://github.com/pandas-dev/pandas/blob/92a52e231534de236c4e878008a4365b4b1da291/ci/code_checks.sh#L144-L415

Currently, some methods fail the GL08 check.

The task here is:

- take 2-4 methods
- run: scripts/validate_docstrings.py --format=actions --errors=GL08 `method-name`
- check if validation docstrings passes for those methods, and if it’s necessary fix the docstrings according to whatever error is reported
- remove those methods from code_checks.sh
- commit, push, open pull request

Please don't comment take as multiple people can work on this issue. You also don't need to ask for permission to work on this, just comment on which methods are you going to work.

If you're new contributor, please check the [contributing guide](https://pandas.pydata.org/docs/dev/development/contributing.html)

**Comment by jordan-d-murphy** [comment]
I don't have permission to add them, but this could probably use some labels:

- CI
- Docs
- good first issue

**Comment by jordan-d-murphy** [comment]
addresses https://github.com/pandas-dev/pandas/pull/57360

**Comment by merlinymy** [comment]
working on      
-      pandas.Index.empty 
-      pandas.Index.names 
-      pandas.Index.view

**Comment by merlinymy** [comment]
working on
     pandas.Index.empty\ 
     pandas.Index.names\ 
     pandas.Index.view\

**Comment by j36426052** [comment]
working on
pandas.IntervalIndex.left\ 
pandas.IntervalIndex.length\ 
pandas.IntervalIndex.mid\

**Comment by jordan-d-murphy** [comment]
Opened a fix for 
```
pandas.DatetimeIndex.as_unit
pandas.DatetimeIndex.freq
```

**Comment by thomasdamcevski** [comment]
Working on
pandas.ExcelFile.book
pandas.MultiIndex.codes

**Comment by lamdang2k** [comment]
Working on 
pandas.Timedelta.microseconds
pandas.Timedelta.unit
pandas.Timedelta.value

**Comment by jordan-d-murphy** [comment]
Opened a fix for `pandas.core.groupby.SeriesGroupBy.value_counts`

**Comment by s1099** [comment]
working on
```
pandas.IntervalIndex.right
pandas.Series.dt
```

**Comment by pmhatre1** [comment]
I am not able to find the proper location.Can anyone help me? I was trying to add docstring for pandas.Period.freq but I am still getting the error.pandas/core/arrays/period.py. I was trying to add docstrig to the 'freq' method.

**Comment by jordan-d-murphy** [comment]
@pmhatre1  I've had the same experience. I could not locate the correct function to add the docstring to. I've encountered this before, and my suspicion is this is something defined in `.pyx` and `.pyi` files and somehow the docstring is invisible to the validations run in `code_checks.sh` If we could figure out a way to solve this, it would allow us to get fixes for so many of the numpy docstring errors failing validation, sadly, at this time I don't have any solution

**Comment by pmhatre1** [comment]
Take      pandas.Series.dt.freq\ 
     pandas.Series.dt.unit\ 
     pandas.Series.empty\

**Comment by pmhatre1** [comment]
> @pmhatre1 I've had the same experience. I could not locate the correct function to add the docstring to. I've encountered this before, and my suspicion is this is something defined in `.pyx` and `.pyi` files and somehow the docstring is invisible to the validations run in `code_checks.sh` If we could figure out a way to solve this, it would allow us to get fixes for so many of the numpy docstring errors failing validation, sadly, at this time I don't have any solution

I think you are right @jordan-d-murphy. Can we pull in someone here to help us?

**Comment by jordan-d-murphy** [comment]
I would love to hear some ideas from folks that might now how to do this. @jrmylow talks about this under [[DOC: fix PR02 errors in docstrings](https://github.com/pandas-dev/pandas/issues/57111#top)](https://github.com/pandas-dev/pandas/issues/57111) in this comment here https://github.com/pandas-dev/pandas/issues/57111#issuecomment-1929460148 maybe we can pursue this?

**Comment by jordan-d-murphy** [comment]
Hi all, if https://github.com/pandas-dev/pandas/pull/57826 gets merged in, I might be reworking our approach here; this would look like closing the following issues: 

https://github.com/pandas-dev/pandas/issues/57443
https://github.com/pandas-dev/pandas/issues/57438
https://github.com/pandas-dev/pandas/issues/57420
https://github.com/pandas-dev/pandas/issues/57417
https://github.com/pandas-dev/pandas/issues/57416
https://github.com/pandas-dev/pandas/issues/57111

And opening a new issue to address these based on the new approach. 

tl;dr 

the work can still be done, but probably under a new ticket once  #57826 is merged in

**Comment by jordan-d-murphy** [comment]
Closed by:
[CI: speedup docstring check consecutive runs #57826](https://github.com/pandas-dev/pandas/pull/57826)
[CI: Better error control in the validation of docstrings #57879](https://github.com/pandas-dev/pandas/pull/57879) 

Opening a new issue to fix docstring errors.

**Comment by jordan-d-murphy** [comment]
Opened [DOC: Enforce Numpy Docstring Validation (Parent Issue) #58063](https://github.com/pandas-dev/pandas/issues/58063) as a parent issue for fixing docstrings based on the refactoring in code_checks.sh 

Feel free to swing by and help out! 🙂
